### ISSUE TITLE
File / code output doesn't work well for translations

### ISSUE THREAD

**Issue by KaiOelfke** [BoltAI/BoltAI#162]
**Describe the bug**

I tried to translate some JSON via base localizations for some iOS app. I added the JSON as document. And edited the system prompt. The first translation for German generated a nice JSON in a copyable box. But subsequent generations resulted in these tokens before.

**To Reproduce**

**Expected behavior**

Have the JSON output as copyable file box.

**Screenshots**

Expected:
<img width="751" alt="Bildschirmfoto 2024-09-15 um 15 04 56" src="https://github.com/user-attachments/assets/1e62f3dc-24f0-462a-bb98-40b6fe7845a2">

Failure:
<img width="740" alt="Bildschirmfoto 2024-09-15 um 15 01 13" src="https://github.com/user-attachments/assets/696ec8c0-7fc6-475d-b050-f592a3ba2fd5">

**Version (please complete the following information):**
 - macOS: 14.6.1
 - app version 1.20.4

**Additional context**

**Comment by KaiOelfke** [comment]
Maybe this happens with languages with special characters due to some encoding bug. French and Finnish for example now worked and Hungarian had the issue again.

**Comment by longseespace** [comment]
No, the LLM just doesn’t always obey your instructions. To fully solve this, we need to use JSON Mode + explicitly tell the LLM to output json. Unfortunately this is not supported in BoltAI yet. 

Maybe create a feature request for that?

https://platform.openai.com/docs/guides/json-mode
